File: Examples/ObjC/Sources/InAppMessageUI/Readme.m
#import "AppDelegate.h"
#import "ReadmeAction.h"
#import "ReadmeViewController.h"

@import BrazeKit;

#pragma mark - Internal

void localSlideup(ReadmeViewController *viewController) {
  BRZInAppMessageRaw *slideup = [[BRZInAppMessageRaw alloc] init];
  slideup.type = BRZInAppMessageRawTypeSlideup;
  slideup.icon = @"";
  slideup.message = @"Local slideup in-app message";
  [AppDelegate.braze.inAppMessagePresenter presentMessage:slideup];
}

void localModal(ReadmeViewController *viewController) {
  BRZInAppMessageRaw *modal = [[BRZInAppMessageRaw alloc] init];
  modal.type = BRZInAppMessageRawTypeModal;
  modal.icon = @"";
  modal.header = @"Header text";
  modal.message = @"Local modal in-app message";
  [AppDelegate.braze.inAppMessagePresenter presentMessage:modal];
}

#pragma mark - Readme

NSString *const readme =
    @"This sample presents how to use the Braze provided in-app message UI:\n"
    @"\n"
    @"- AppDelegate.{h,m}\n"
    @"  - In-app message UI configuration\n"
    @"  - In-app message UI delegate\n"
    @"- BRZGIFViewProvider+SDWebImage.{h,m}\n"
    @"  - Use SDWebImage to provide GIF support to the Braze UI components";

NSInteger const actionsCount = 2;

ReadmeAction const actions[] = {
    {@"Present local slideup in-app message", @"",
     ^(ReadmeViewController *_Nonnull viewController){
         localSlideup(viewController);
}
}
, {@"Present local modal in-app message", @"",
   ^(ReadmeViewController *_Nonnull viewController){
       localModal(viewController);
}
}
,
}
;

File: Examples/InAppMessages/Readme.swift
import UIKit
import BrazeKit

let readme =
  """
  This sample presents how to use the Braze provided in-app message UI:

  - AppDelegate.swift:
    - In-app message UI configuration
    - In-app message UI delegate
  - SDWebImageGIFViewProvider.swift:
    - Use SDWebImage to provide gif support to the in-app message UI
  """

let actions: [(String, String, (ReadmeViewController) -> Void)] = [
  (
    "Present local slideup in-app message",
    "",
    localSlideup
  ),
  (
    "Present local modal in-app message",
    "",
    localModal
  )
]

// MARK: - Internal

func localSlideup(_ viewController: ReadmeViewController) {
  let slideup: Braze.InAppMessage = .slideup(
    .init(
      graphic: .icon(""),
      message: "Local slideup in-app message"
    )
  )
  AppDelegate.braze?.inAppMessagePresenter?.present(message: slideup)
}

func localModal(_ viewController: ReadmeViewController) {
  let modal: Braze.InAppMessage = .modal(
    .init(
      graphic: .icon(""),
      header: "Header text",
      message: "Local modal in-app message"
    )
  )
  AppDelegate.braze?.inAppMessagePresenter?.present(message: modal)
}

